domain-services/ifix-master-data-service/src/main/java/org/egov/repository/ServiceRequestRepository.java

```diff
@@ -0,0 +1,45 @@
+package org.egov.repository;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.tracer.model.ServiceCallException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Repository
+@Slf4j
+public class ServiceRequestRepository<R,T> {
```

Comment:
I don't think it's good to define R and T while creating an object of this. Take for example, in project util we will have to write api call to government, eat and department. If we follow this approach we will have to make separate objects for each. 
So I suggest let the request and response be generic object as kept in PGR service.

Reply:
But when we are returning the response as Object like in PGR service. There is overhead of typecasting to check the json attributes values.

Reply:
Let there be that overhead. And when the services are separated out in different projects, every service will not define Government or Department models. They will only form a json request body.

Reply:
done..